Tested on mingw64, cpp-test finished normally. Building on mingw32 still failed due to `__stdcall` mangling on libglew32.a (undefined reference to 'glewInit'), which I'm trying to fix. (I don't have Visual Studio).

> A DWORD_PTR is an unsigned long type used for pointer precision. It is used when casting a pointer to an unsigned long type to perform pointer arithmetic. DWORD_PTR is also commonly used for general 32-bit parameters that have been extended to 64 bits in 64-bit Windows. 

From what MSDN said, DWORD_PTR is 32-bit on Win32, which is the same as DWORD, so Win32 should be fine. When casting LPVOID to DWORD on Win64, gcc failed to compile and reported an error. I don't know whether Visual Studio will report an error or warning, or just fix it for us.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -153,7 +153,7 @@ include_directories(
   ${CMAKE_CURRENT_SOURCE_DIR}/external/xxhash
 )
 
-if(WIN32)
+if(WIN32 AND NOT MINGW)
   include_directories(
     ${CMAKE_CURRENT_SOURCE_DIR}/external/${PLATFORM_FOLDER}-specific/gles/include/OGLES
     ${CMAKE_CURRENT_SOURCE_DIR}/external/${PLATFORM_FOLDER}-specific/icon/include

--- a/tests/cpp-empty-test/CMakeLists.txt
+++ b/tests/cpp-empty-test/CMakeLists.txt
@@ -73,3 +73,7 @@ else()
 endif()
 
 target_link_libraries(${APP_NAME} audio cocos2d)
+# MinGW builds need to link libws2_32 and libglew32 manually
+if (MINGW)
+  target_link_libraries(${APP_NAME} glew32 ws2_32)
+endif()

--- a/tests/cpp-tests/CMakeLists.txt
+++ b/tests/cpp-tests/CMakeLists.txt
@@ -205,6 +205,10 @@ target_link_libraries(${APP_NAME}
   cocos2d
   box2d
 )
+# MinGW builds need to link libws2_32 manually
+if (MINGW)
+  target_link_libraries(${APP_NAME} ws2_32)
+endif()
 
 set(APP_BIN_DIR "${CMAKE_BINARY_DIR}/bin/${APP_NAME}")
 

